main.go

```diff
@@ -185,3 +206,34 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func createClient() (*http.Client, error) {
+	certDirFormat := "/etc/osbuild/certs/%s"
+	ca := fmt.Sprintf(certDirFormat, "ca.crt")
+	tlsCert := fmt.Sprintf(certDirFormat, "tls.crt")
+	tlsKey := fmt.Sprintf(certDirFormat, "tls.key")
+	var tlsConfig *tls.Config
+
+	caCert, err := ioutil.ReadFile(filepath.Clean(ca))
+	if err != nil {
+		return nil, err
+	}
+
+	cert, err := tls.LoadX509KeyPair(tlsCert, tlsKey)
+	if err != nil {
+		return nil, err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig = &tls.Config{
+		MinVersion:   tls.VersionTLS12,
+		RootCAs:      caCertPool,
+		Certificates: []tls.Certificate{cert},
+	}
+
+	tlsConfig.BuildNameToCertificate()
```

Comment:
I think this call is deprecated: https://pkg.go.dev/crypto/tls#Config.BuildNameToCertificate